python/tank/platform/interactive_login.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+UI and console based login for Toolkit.
+"""
+
+from getpass import getpass
+import logging
+
+from tank_vendor.shotgun_api3 import Shotgun, AuthenticationFault
+from tank.util.session import cache_session_data, get_cached_login_info, delete_session_data, is_session_token_cached
+from tank.util.login import get_login_name
+from tank.util.shotgun import get_associated_sg_config_data, get_associated_sg_base_url
+
+# Configure logging
+logger = logging.getLogger("sgtk-interactive_login")
+# logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler())
+
+
+def _get_session_token(hostname, login, password, http_proxy):
+    """
+    Retrieves a session token for the given credentials.
+    :param hostname: The host to connect to.
+    :param login: The user to get a session for.
+    :param password: Password for the user.
+    :param http_proxy: Proxy to use. Can be None.
+    :returns: If the credentials were valid, returns a session token, otherwise returns None.
+    """
+    try:
+        # Create the instance...
+        sg = Shotgun(
+            hostname.encode("utf-8"),
+            login=login.encode("utf-8"),
+            password=password.encode("utf-8"),
+            http_proxy=http_proxy.encode("utf-8") if http_proxy else None
+        )
+        # .. and generate the session token. If it throws, we have invalid credentials.
+        return sg.get_session_token()
+    except AuthenticationFault:
+        print "Authentication failed."
+        return None
+    except:
+        # We couldn't login, so try again.
+        logging.exception("There was a problem logging in.")
+        return None
+
+
+def _do_ui_based_login(hostname, login, http_proxy):
+    pass
+
+
+def _get_keyboard_input(label, default_value=""):
```

Comment:
The code you are referring to in console_utils is meant for Yes/No/All questions only. Maybe there could be a way to refactor this however to provide common behaviour and presentation for keyboard inputs. Let’s discuss this offline in our next meeting. I’ll add this to the Hot Topics.

From: Manne Öhrström <notifications@github.com<mailto:notifications@github.com>>
Reply-To: shotgunsoftware/tk-core <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, February 17, 2015 at 9:22 AM
To: shotgunsoftware/tk-core <tk-core@noreply.github.com<mailto:tk-core@noreply.github.com>>
Cc: JF Boismenu <jean.francois.boismenu@autodesk.com<mailto:jean.francois.boismenu@autodesk.com>>
Subject: Re: [tk-core] For #28030, [security] User/password sign in on tank command with session caching (#129)


In python/tank/platform/interactive_login.py<https://github.com/shotgunsoftware/tk-core/pull/129#discussion_r24818099>:

> +        # .. and generate the session token. If it throws, we have invalid credentials.
> +        return sg.get_session_token()
> +    except AuthenticationFault:
> +        print "Authentication failed."
> +        return None
> +    except:
> +        # We couldn't login, so try again.
> +        logging.exception("There was a problem logging in.")
> +        return None
> +
> +
> +def _do_ui_based_login(hostname, login, http_proxy):
> +    pass
> +
> +
> +def _get_keyboard_input(label, default_value=""):


hey, check out tank.deploy.tank_commands.console_utils.py - I think there may be a method like this which can be reused!

—
Reply to this email directly or view it on GitHub<https://github.com/shotgunsoftware/tk-core/pull/129/files#r24818099>.